sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -397,11 +414,44 @@ public Read usingStandardSql() {
       return toBuilder().setUseLegacySql(false).build();
     }
 
+    /**
+     * Use new template-compatible source implementation.
+     *
+     * <p>Use new template-compatible source implementation. This implementation is
+     * compatible with repeated template invocations. It does not support dynamic work
+     * rebalancing.
+     */
+    @Experimental(Experimental.Kind.SOURCE_SINK)
+    public Read withNewSource() {
```

Comment:
I still think that "withNewSource" is not a good name for this method: if any user that wants to use BigQueryIO.read() with templates must call this method, then the method should be called withTemplateCompatibility() or something like that.

Also, would be good if the comment explained why we can't automatically call this if the job is being run from a template and why the user has to do it instead.